src/Components/AdvancedSearch/Internal/AdvancedSearchResultDetailsList/AdvancedSearchResultDetailsList.tsx

```diff
@@ -0,0 +1,114 @@
+    IconButton,
+    SelectionMode
+} from '@fluentui/react';
+import { IADTTwin } from '../../../../Models/Constants';
+import { useTranslation } from 'react-i18next';
+const getClassNames = classNamesFunction<
+    IAdvancedSearchResultDetailsListStyleProps,
+    IAdvancedSearchResultDetailsListStyles
+>();
+
+const AdvancedSearchResultDetailsList: React.FC<IAdvancedSearchResultDetailsListProps> = ({
+    filteredTwinsResult,
+    additionalColumns,
+    callbackFunction,
+    styles,
+    adapter,
+    theme,
+    locale
+}) => {
+    const { t } = useTranslation();
+
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+    const columns: IColumn[] = [
+        {
+            key: `twin-id + ${Math.random()}`,
+            name: t('twinId'),
+            fieldName: '$dtId',
+            minWidth: 100,
+            maxWidth: 200,
+            isResizable: true
+        },
+        {
+            key: 'properties',
+            name: t('properties'),
+            minWidth: 100,
+            maxWidth: 200,
+            isResizable: true
+        }
+    ];
+    const renderItemColumn: IDetailsListProps['onRenderItemColumn'] = (
+        item: any,
+        _itemIndex: number,
+        column: IColumn
+    ) => {
+        const fieldContent = String(item[column.fieldName]);
+        switch (column.key) {
+            case 'properties':
+                return (
+                    <IconButton
+                        iconProps={{ iconName: 'EntryView' }}
+                        title={'inspect property'}
+                        ariaLabel={'inspect property'}
+                        onClick={(event) => {
+                            event.stopPropagation();
+                            alert('TO-DO, make property inspector show up');
+                        }}
+                        className={'cb-scenes-action-button'}
+                    />
+                );
+            default:
+                return <span>{fieldContent}</span>;
+        }
+    };
+
+    const onItemInvoked = useCallback((twin: IADTTwin) => {
+        alert(`Twin Id ${twin.$dtId} was selected`);
+    }, []);
+    return (
+        <div className={classNames.root}>
+            <h1>{'Results (' + filteredTwinsResult.length + ')'}</h1>
+            <section>
+                <DetailsList
+                    items={filteredTwinsResult}
+                    columns={
+                        additionalColumns.length > 0
+                            ? columns.concat(additionalColumns)
+                            : columns
+                    }
+                    layoutMode={DetailsListLayoutMode.fixedColumns}
+                    selectionPreservedOnEmptyClick={true}
+                    ariaLabelForSelectionColumn="Toggle selection"
+                    ariaLabelForSelectAllCheckbox="Toggle selection for all items"
+                    checkButtonAriaLabel="select row"
+                    onItemInvoked={onItemInvoked}
```

Comment:
for some reason clicking anywhere on a row does the selection, but then clicking the same spot on the row will not unselect, in my mind, you should be able to select/unselect by clicking anywhere on the row. Is that possible to do?